pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/name/SegmentNameGenerator.java

```diff
@@ -37,7 +37,7 @@ public interface SegmentNameGenerator extends Serializable {
    * @param segmentName provide segment name
    * @return true if segmentName is valid.
    */
-  default boolean isValidSegmentName(String segmentName) {
+  static boolean isValidSegmentName(String segmentName) {
```

Comment:
Let's create a `SegmentNameUtils` class and move this method there

Reply:
Consider changing it to `void validateSegmentName(String segmentName)` and throw `IllegalArgumentException` to simplify the caller code.

Reply:
thanks for the feedback, updated the PR.